tests/test_sshfsmount.cpp

```diff
@@ -369,11 +413,50 @@ TEST_F(SshfsMount, works_with_absolute_paths)
 {
     CommandVector commands = {
         {"sudo /bin/bash -c 'P=\"/home/ubuntu/target\"; while [ ! -d \"$P/\" ]; do P=${P%/*}; done; echo $P/'",
-         "/home/ubuntu/"},
-        {"id -u", "1000"},
-        {"id -g", "1000"},
-        {"id -u", "1000"},
-        {"id -g", "1000"}};
+         "/home/ubuntu/"}};
 
     test_command_execution(commands, std::string("/home/ubuntu/target"));
 }
+
+TEST_F(SshfsMount, throws_install_sshfs_which_snap_fails)
+{
+    bool invoked{false};
+    auto request_exec = make_exec_that_fails_for({"which snap"}, invoked);
+    REPLACE(ssh_channel_request_exec, request_exec);
+
+    mp::SSHSession session{"a", 42};
+
+    EXPECT_THROW(mp::utils::install_sshfs_for("foo", session), std::runtime_error);
+    EXPECT_TRUE(invoked);
+}
+
+TEST_F(SshfsMount, throws_install_sshfs_no_snap_dir_fails)
+{
+    bool invoked{false};
+    auto request_exec = make_exec_that_fails_for({"[ -e /snap ]"}, invoked);
+    REPLACE(ssh_channel_request_exec, request_exec);
+
+    mp::SSHSession session{"a", 42};
+
+    EXPECT_THROW(mp::utils::install_sshfs_for("foo", session), std::runtime_error);
+    EXPECT_TRUE(invoked);
+}
+
+TEST_F(SshfsMount, throws_install_sshfs_snap_install_fails)
+{
+    bool invoked{false};
+    auto request_exec = make_exec_that_fails_for({"sudo snap install multipass-sshfs"}, invoked);
+    REPLACE(ssh_channel_request_exec, request_exec);
+
+    mp::SSHSession session{"a", 42};
+
+    EXPECT_THROW(mp::utils::install_sshfs_for("foo", session), mp::SSHFSMissingError);
+    EXPECT_TRUE(invoked);
+}
+
+TEST_F(SshfsMount, install_sshfs_no_failures_does_not_throw)
+{
+    mp::SSHSession session{"a", 42};
+
+    EXPECT_NO_THROW(mp::utils::install_sshfs_for("foo", session));
+}
```

Comment:
This should probably be a bit more verbose, checking that the expected commands are executed (using `test_command_execution`).

Reply:
Also, missing case for the timeout, if possible to test?

Reply:
And a couple more untested cases:

https://codecov.io/gh/canonical/multipass/compare/e4cab2e0ade0ebdb04e51d770da6a7fed224bce5...c5c9f0c2df427023c788eabc16f2ebbe534b74c2/diff#D9-101

Reply:
Ack

Reply:
For the timeout, I can add a timeout parameter to the function with a default of 5 minutes that we can override in the test.

Reply:
There's no way to mock the timeout passing regardless of the actual timeout? I'd rather avoid an actual timeout expiring in the tests :)

Reply:
No, we have no mocks in place for timeouts of any sort.  But like we do in other tests, we can put in a very short timeout and then wait for that.

Reply:
@Saviq,

Looking at this particular test some more (`install_sshfs_no_failures_does_not_throw`), we can't easily use `test_command_execution` since all that is being tested here is `mp::utils::install_sshfs_for()`.  `test_command_execution` creates a whole mock `SshfsServer` whereas this test is not.  Unless I refactor `test_command_execution` in this PR, I think it should wait and this test is the best we can do.

I will still try to address the timeout test along with other couple of missing coverage pieces.

Reply:
I still have the timeout test to try to address.